src/SprykerEco/Zed/Unzer/Communication/Plugin/Payment/UnzerIntersectionPaymentMethodFilterPlugin.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Unzer\Communication\Plugin\Payment;
+
+use Generated\Shared\Transfer\PaymentMethodsTransfer;
+use Generated\Shared\Transfer\QuoteTransfer;
+use Spryker\Zed\Kernel\Communication\AbstractPlugin;
+use Spryker\Zed\PaymentExtension\Dependency\Plugin\PaymentMethodFilterPluginInterface;
+
+/**
+ * @method \SprykerEco\Zed\Unzer\Business\UnzerFacadeInterface getFacade()
+ * @method \SprykerEco\Zed\Unzer\Communication\UnzerCommunicationFactory getFactory()
+ * @method \SprykerEco\Zed\Unzer\UnzerConfig getConfig()
+ */
+class UnzerIntersectionPaymentMethodFilterPlugin extends AbstractPlugin implements PaymentMethodFilterPluginInterface
+{
+    /**
+     * {@inheritDoc}
+     * - Takes allowed payment methods from the `QuoteTransfer` received from Unzer local config.
```

Comment:
I assume required transfer properties are not listed
please check same for called facade method